HotelsProject/Modules/DataAccessLogic.py

```diff
@@ -0,0 +1,121 @@
+from Generals.DBOperationType import DBOperationType
+from Generals.SQLHandler import run_db_operation
+from DataObjects.Reservation import Reservation
+from DataObjects.HotelRoom import HotelRoom
+from DataObjects.HotelRoomInventory import HotelRoomInventory
+from datetime import date
+
+
+async def get_reservation_by_id(reservation_id):
+    """
+    Queries reservation by given id
+    :param reservation_id:
+    :return: Reservation object
+    """
+    reservation = None
+    reservation_query = """SELECT t1.reservation_id, t1.room_id, t2.hotel_id,
+                                  t2.room_type_name, t1.arrival_date, t1.departure_date, t1.status 
+                            FROM reservations t1 join 
+                                 hotel_rooms t2 on t1.room_id = t2.room_id 
+                            WHERE reservation_id = $1"""
+    query_result = await run_db_operation(DBOperationType.QUERY_SINGLE, reservation_query, int(reservation_id))
+    if query_result:
+        reservation = Reservation(query_result['reservation_id'],
+                                  query_result['hotel_id'],
+                                  query_result['room_id'],
+                                  query_result['room_type_name'],
+                                  str(query_result['arrival_date']),
+                                  str(query_result['departure_date']),
+                                  query_result['status'])
+    return reservation
+
+
+async def update_reservation_status(reservation_id, status):
```

Comment:
Maybe it should return a boolean instead of an int :-) If the operation was successful

Reply:
It returns the number of rows updated. Later, in Business Logic, it checks this number and raises an error if no rows were updated.